Setup and loading

BenjaminArp opened the pull request:
This is the initial setup of our ILP approach.

It successfully implements a Mitigation for a minimal DFD.
Currently, a mitigation strategy needs to be added; later, this will be automated.

BenjaminArp committed:
Inital setup

BenjaminArp committed:
Added file output for optimization problem

BenjaminArp committed:
Add google ILP solver

BenjaminArp committed:
add assertion to testcase

BenjaminArp committed:
add Automized Mitigation extraction aswell as Tuhh models

BenjaminArp committed:
add gateway constraint

BenjaminArp committed:
add efficency test

BenjaminArp committed:
formatter

BenjaminArp committed:
New ModelCostCalculater that ignores forwarding

uuqjz commented on bundles/dev.arcovia.mitigation.ilp/src/dev/arcovia/mitigation/ilp/Constraint.java:
```diff
@@ -0,0 +1,71 @@
+        var translation = new CNFTranslation(dsl);
+
+        var literals = translation.constructCNF()
+                .get(0)
```
Same here.

BenjaminArp replied:
Because constructToCNF always returns a list with just  1 element, to get the literals of the cnf i need to extract it from the list first.

uuqjz commented on bundles/dev.arcovia.mitigation.ilp/src/dev/arcovia/mitigation/ilp/Constraint.java:
```diff
@@ -0,0 +1,71 @@
+                .get(0)
+                .literals();
+
+        List<MitigationStrategy> mit = new ArrayList<>();
```
No abbreviations please

uuqjz commented on bundles/dev.arcovia.mitigation.ilp/src/dev/arcovia/mitigation/ilp/Node.java:
```diff
@@ -0,0 +1,122 @@
+                switch (mitigation.type) {
+                    case Node -> {
+                        List<Mitigation> req = new ArrayList<>();
+                        for (var r : mitigation.required)
```
what does r mean? please use speaking names

uuqjz commented on bundles/dev.arcovia.mitigation.ilp/src/dev/arcovia/mitigation/ilp/OptimizationManager.java:
```diff
@@ -0,0 +1,300 @@
+    private List<Term> getActions(List<Mitigation> result) {
+        List<Mitigation> additoonal = new ArrayList<>();
+        for (var mit : result) {
+            additoonal.addAll(mit.required());
```
spelling

uuqjz commented on tests/dev.arcovia.mitigation.ilp.tests/models/mindfd.json:
```diff
@@ -0,0 +1,105 @@
+{
```
is this a copy from the sat test bundle? if yes use the original dfd and do not make a copy here.

uuqjz commented on tests/dev.arcovia.mitigation.ilp.tests/models/mindfd-repaired.json:
```diff
@@ -0,0 +1,149 @@
+{
```
same thing about stage repaired jsons

uuqjz commented on tests/dev.arcovia.mitigation.ranking.tests/meassurement_results.txt:
```diff
@@ -2026,3 +2026,161 @@ RAW: 1215.0
 RAW: 1348.0
 RAW: 1144.0
 jferrater_INCREASING: 1279.375
+RAW: 940.0
```
there should be no changes here, please restore original file

BenjaminArp commented on bundles/dev.arcovia.mitigation.ilp/src/dev/arcovia/mitigation/ilp/Constraint.java:
```diff
@@ -0,0 +1,71 @@
+    public boolean isPrecondition(CompositeLabel label) {
+        var translation = new CNFTranslation(dsl);
+        var literals = translation.constructCNF()
+                .get(0)
```
Because constructToCNF always returns a list with just  1 element, to get the literals of the cnf i need to extract it from the list first.

uuqjz replied:
returning a list of 1 is odd. is it always just one element, or just in your case?

BenjaminArp replied:
i could not find a case where it is more then one element

BenjaminArp commented on bundles/dev.arcovia.mitigation.sat/src/dev/arcovia/mitigation/sat/ModelCostCalculator.java:
```diff
@@ -69,6 +72,49 @@ public int calculateCost() {
 
         return cost;
     }
+    public int calculateCostWithoutForwarding() {
```
thats the plan

BenjaminArp commented on pom.xml:
```diff
@@ -11,6 +11,22 @@
 	<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
  </properties>
 
+ <dependencyManagement>
```
According to nicolas those changes are needed for maven.

BenjaminArp committed:
Implement Feedback

- Mainly renaming & spelling fixes

- Extracted requiredMitgation Logic to its own function, and fixing the limitation of only allowing a mitigation of just one level of required mitgations by using recursion

BenjaminArp committed:
remove testresults

BenjaminArp committed:
use default model from sat

BenjaminArp committed:
run formatter
